src/pyhf/contrib/viz/brazil.py

```diff
@@ -27,24 +27,124 @@ def plot_results(ax, mutests, tests, test_size=0.05):
+        >>> data = observations + model.config.auxdata
+        >>> poi_vals = np.linspace(0, 5, 41)
+        >>> results = [
+        ...     pyhf.infer.hypotest(
+        ...         test_poi, data, model, return_expected_set=True, return_tail_probs=True
+        ...     )
+        ...     for test_poi in poi_vals
+        ... ]
+        >>> fig, ax = plt.subplots()
+        >>> pyhf.contrib.viz.brazil.plot_cls_components(ax, poi_vals, results)
+
+    Args:
+        ax (:obj:`matplotlib.axes.Axes`): The matplotlib axis object to plot on.
+        mutests (:obj:`list` or :obj:`array`): The values of the POI where the
+          hypothesis tests were performed.
+        tests (:obj:`list` or :obj:`array`): The collection of :math:`p`-values
+          from the hypothesis tests.
+          ``tests`` is required to have the same structure as
+          :func:`pyhf.infer.hypotest`'s return when using ``return_expected_set=True``
+          and ``return_tail_probs=True``: a tuple of :math:`\mathrm{CL}_{s}`,
+          :math:`\left[\mathrm{CL}_{s+b}, \mathrm{CL}_{b}\right]`,
+          :math:`\mathrm{CL}_{s,\mathrm{exp}}` band.
+        test_size (:obj:`float`): The size, :math:`\alpha`, of the test.
+        clb_only (:obj:`bool`): Bool for plotting only the :math:`\mathrm{CL}_{b}`
+          component.
+        clsb_only (:obj:`bool`): Bool for plotting only the :math:`\mathrm{CL}_{s+b}`
+          component.
+    """
+
+    if len(tests[0]) != 3:
+        raise ValueError(
+            f"The components of 'tests' should have len of 3 but have len {len(tests[0])}."
+            + "\n'tests' should have format of: CLs_obs, [CLsb, CLb], [CLs_exp band]"
+        )
+
+    # split into components
+    CLs_obs = np.array([test[0] for test in tests])
+    tail_probs = np.array([test[1] for test in tests])
+    CLs_exp_set = np.array([test[2] for test in tests])
+
+    # zip CLs_obs and CLs_exp_set back into format for plot_results
+    CLs_results = [(obs, exp_set) for obs, exp_set in zip(CLs_obs, CLs_exp_set)]
+
+    # plot CLs_obs and CLs_expected set
+    y_label = kwargs.pop("ylabel", r"$p\,$-value")
+    CLs_color = kwargs.pop("cls_color", "black")
+    plot_results(ax, mutests, CLs_results, test_size, ylabel=y_label, color=CLs_color)
+
+    CLsb_obs = np.array([tail_prob[0] for tail_prob in tail_probs])
+    CLb_obs = np.array([tail_prob[1] for tail_prob in tail_probs])
+
+    linewidth = kwargs.pop("linewidth", 2)
+    if not clb_only:
+        CLsb_color = kwargs.pop("clsb_color", "red")
+        ax.plot(
+            mutests,
+            CLsb_obs,
+            color=CLsb_color,
+            linewidth=linewidth,
+            label=r"$\mathrm{CL}_{s+b}$",
+        )
+    if not clsb_only:
+        CLb_color = kwargs.pop("clb_color", "blue")
+        ax.plot(
+            mutests,
+            CLb_obs,
+            color=CLb_color,
+            linewidth=linewidth,
+            label=r"$\mathrm{CL}_{b}$",
+        )
+    ax.legend(loc="best")
```

Comment:
If you add a legend, I would suggest also including the expected CLs values and uncertainties in there, and possibly even the CLs=0.05 line.

Reply:
Good point. Time to bust out the `matplotlib` docs on how patches work again. :P